src/utils/createGenerateClassName.js

```diff
@@ -15,5 +21,23 @@ const jssCounter = globalRef[namespace]++
 export default (): generateClassName => {
   let ruleCounter = 0
 
-  return (rule: Rule): string => `${rule.key}-${jssCounter}-${ruleCounter++}`
+  return (rule: Rule, sheet?: StyleSheet): string => {
+    ruleCounter += 1
+
+    if (ruleCounter > maxRules) {
+      warning(
+        false,
+        'You might have a memory leak. Rule counter is at %s.',
+        ruleCounter
+      )
+    }
+
+    if (env === 'production') {
+      return `c${jssCounter}${ruleCounter}`
```

Comment:
@kof Is right, we have examples in the documentation of Material-UI. No need to fork. Also given the edge case you are facing I would vote for people like you creating their own `createGenerateClassName` function, at their own risk.